### ISSUE TITLE
bug: cannot use wildcard in apigroup and resources (and just use label selectors) in EdgePlacement

### ISSUE THREAD

**Issue by pdettori** [kubestellar/kubestellar#1498]
### Describe the bug

I would expect to be able to just specify labels for objects to downsync in the edge placement, rather than enumerating all api groups and resources for all. 

For example, I would expect the following edge placement to work with any object with label `app=foo`

```
apiVersion: edge.kubestellar.io/v2alpha1
kind: EdgePlacement
metadata:
  name: ep2
spec:
  locationSelectors:
  - matchLabels: {"location-group":"edge"}
  downsync:
  - apiGroup: "*"
    resources: [ "*" ]
    namespaces: [ "*" ]
    objectNames: [ "*" ]
    labelSelectors:
    - matchLabels: {"app":"foo"}
EOF
``` 

### Steps To Reproduce

1. Follow the steps in KubeStellar Quickstart guide using kind (https://docs.kubestellar.io/release-0.14/Getting-Started/user-quickstart-kind/ ) up to step 4.
2. Apply the following placement:
```
KUBECONFIG=ks-core.kubeconfig kubectl apply -f - <<EOF
apiVersion: edge.kubestellar.io/v2alpha1
kind: EdgePlacement
metadata:
  name: ep2
spec:
  locationSelectors:
  - matchLabels: {"location-group":"edge"}
  downsync:
  - apiGroup: "*"
    resources: [ "*" ]
    namespaces: [ "*" ]
    objectNames: [ "*" ]
    labelSelectors:
    - matchLabels: {"app":"foo"}
EOF
```
3. Apply the following deployment:
```
KUBECONFIG=ks-core.kubeconfig kubectl apply -f - <<EOF
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: default
  name: ks-deployment2
  labels:
    app: foo
spec:
  selector: {matchLabels: {app: common} }
  template:
    metadata:
      labels: {app: common}
    spec:
      containers:
      - name: httpd
        image: public.ecr.aws/docker/library/httpd:2.4
        ports:
        - name: http
          containerPort: 80
          hostPort: 8089
          protocol: TCP
        volumeMounts:
        - name: htdocs
          readOnly: true
          mountPath: /usr/local/apache2/htdocs
      volumes:
      - name: htdocs
        configMap:
          name: httpd-htdocs
          optional: false
EOF
```
4. Verify that no deployments are created on  ks-edge-cluster1 or  ks-edge-cluster2
```
KUBECONFIG=~/.kube/config kubectl --context ks-edge-cluster1 get deployments -A
KUBECONFIG=~/.kube/config kubectl --context ks-edge-cluster2 get deployments -A
```

### Expected Behavior

I would expect to see deployment with label app=foo propagated to ks-edge-cluster1 and ks-edge-cluster2

### Additional Context

To make this scenario work, I had to specify the api group and resources:
```
KUBECONFIG=ks-core.kubeconfig kubectl apply -f - <<EOF
apiVersion: edge.kubestellar.io/v2alpha1
kind: EdgePlacement
metadata:
  name: ep2
spec:
  locationSelectors:
  - matchLabels: {"location-group":"edge"}
  downsync:
  - apiGroup: "apps"
    resources: [ "deployments" ]
    namespaces: [ "*" ]
    objectNames: [ "*" ]
    labelSelectors:
    - matchLabels: {"app":"foo"}
EOF
```
However, this still requires enumerating the api groups/resources, which might be unpractical in some scenarios - e.g. installing a helm chart with many different api resources inside - I would need to look inside the helm chart to see what is there and build the edge placement for that. Also, this might be an issue if the helm chart publishers add other api resources in the chart - upgrading the chart may become operationally more complex.

**Comment by MikeSpreitzer** [comment]
Versions have never been part of the downsync or upsync descriptions.

The API Group was intended to be optional, as you can tell in the opening comment of #1202: "The main idea is to make DownsyncObjectTest a simple conjunction of tests, with each field defaulting to matching everything". There was one oversight in that PR: marking the APIGroup field as optional. This was fixed in #1297, which merged after release 0.14.0.

**Comment by pdettori** [comment]
@MikeSpreitzer , thanks
> Versions have never been part of the downsync or upsync descriptions.

You are correct, I meant "resources" instead of version, and updated the issue accordingly

> This was fixed in https://github.com/kubestellar/kubestellar/pull/1297, which merged after release 0.14.0.

I tested on 0.14.0 as the 0.15.0 is based on the decoupled architecture and I was not able to run it yet. But thanks for the clarification, it's good that it's already implemented.

**Comment by pdettori** [comment]
Closing based on @MikeSpreitzer comment above/
